core/node/provider.go

```diff
@@ -59,29 +62,78 @@ func SimpleProviderSys(isOnline bool) interface{} {
 	}
 }
 
+type provideMany interface {
+	ProvideMany(ctx context.Context, keys []multihash.Multihash) error
+	Ready() bool
+}
+
+// BatchedProviderSys creates new provider system
+func BatchedProviderSys(isOnline bool, reprovideInterval string) interface{} {
+	return func(lc fx.Lifecycle, cr libp2p.BaseIpfsRouting, q *q.Queue, keyProvider simple.KeyChanFunc, repo repo.Repo) (provider.System, error) {
+		r, ok := (cr).(provideMany)
+		if !ok {
+			return nil, fmt.Errorf("BatchedProviderSys requires a content router that supports provideMany")
+		}
+
+		reproviderInterval := kReprovideFrequency
```

Comment:
minor: Please rename `reproviderInterval` to `reprovideDuration`.  Otherwise, it is difficult to distinguish from the `reprovideInterval` parameter.

Reply:
thx for the catch, that's definitely confusing to look at. I changed it to `reprovideIntervalDuration` since `reprovideDuration` sounds like how long a reprovide takes as opposed to the gap between reprovides. Maybe it's a little bit of a long name, but 🤷.